pkg/controller/persistentvolume/types.go

```diff
@@ -105,19 +115,29 @@ func (pvIndex *persistentVolumeOrderedIndex) Find(searchPV *api.PersistentVolume
 		// return the exact pre-binding match, if found
 		unboundVolumes := []*api.PersistentVolume{}
 		for _, volume := range volumes {
-			// check for current binding
 			if volume.Spec.ClaimRef != nil {
+				if preboundClaim != "" {
+					boundClaim := fmt.Sprintf("%s/%s", volume.Spec.ClaimRef.Namespace, volume.Spec.ClaimRef.Name)
+					if boundClaim == preboundClaim {
+						// exact match! No search required.
+						return volume, nil
+					}
+				}
 				continue
 			}
 
-			// check for pre-bind where the volume is intended for one specific claim
-			if createdFor, ok := volume.Annotations[createdForKey]; ok {
-				if createdFor != searchPV.Annotations[createdForKey] {
-					// the volume is pre-bound and does not match the search criteria.
-					continue
+			// TODO: can probably remove this entirely.  The annotation is "experimental" and now duplicates
+			// the above check for ClaimRef
+			if preboundClaim != "" {
```

Comment:
This change in behavior makes the assumption here: https://github.com/kubernetes/kubernetes/pull/16432/files#diff-113bc1e85667b5dff1e24729f402a4b9R143 invalid.